app/helpers/session_timeout_warning_helper.rb

```diff
@@ -11,6 +11,13 @@ def warning
     (Figaro.env.session_timeout_warning_seconds || 30).to_i
   end
 
+  def timeout_refresh_url
+    URI(request.original_url).tap do |url|
+      query = Rack::Utils.parse_nested_query(url.query).with_indifferent_access
+      url.query = query.merge(timeout: true).to_query
+    end.to_s.html_safe # rubocop:disable Rails/OutputSafety
+  end
```

Comment:
Nothing off the top of my head besides parsing the HTML to see if there is a form. I believe the only offline pages we currently have without a form are the `/privacy` and `/sign_up/start` pages.